scripts/src/scheduler.js

```diff
@@ -0,0 +1,91 @@
+require('console-stamp')(console, '[HH:MM:ss.l]')
+const { ContractPromise } = require('@polkadot/api-contract')
+const { ApiPromise, WsProvider, Keyring } = require('@polkadot/api')
+const PhalaSdk = require('@phala/sdk')
+const PhalaSDKTypes = PhalaSdk.types
+const KhalaTypes = require('@phala/typedefs').khalaDev
+const fs = require('fs')
+const path = require('path')
+const express = require('express')
+
+const NODE_ENDPOINT = 'wss://poc5.phala.network/ws'
+const PRUNTIME_ENDPOINT = 'https://poc5.phala.network/tee-api-1'
+const CONTRACT_ID = '0x90a1cbf2a00c76e16d53cd3568c639eacbb30076138cf38270e4673f37c6a3ff'
+const EXE_WORKER = '0x2eaaf908adda6391e434ff959973019fb374af1076edd4fec55b5e6018b1a955'
+const SOURCE = 'Moonbeam'
+
+function loadContractFile(contractFile) {
+    const metadata = JSON.parse(fs.readFileSync(contractFile, 'utf8'))
+    const constructor = metadata.V3.spec.constructors.find(
+      c => c.label == 'new',
+    ).selector
+    const name = metadata.contract.name
+    const wasm = metadata.source.wasm
+    return { wasm, metadata, constructor, name }
+}
+
+async function createContract(api, pruntimeUrl, contract, contractID) {
+    const { api: workerApi } = await PhalaSdk.create({
+      api,
+      baseURL: pruntimeUrl,
+      contractId: contractID,
+      autoDeposit: true,
+    })
+    const contractApi = new ContractPromise(
+      workerApi,
+      contract.metadata,
+      contractID,
+    )
+    return contractApi
+}
+
+async function main() {
+    console.log(`Loading contract metedata form file system`)
+    const contract = loadContractFile(
+        path.join(__dirname, '../../target/ink/index_executor/index_executor.contract'),
+    )
+
+    console.log(`Establishing connection with blockchain node`)
+    const nodeApi = await ApiPromise.create({
+        provider: new WsProvider(NODE_ENDPOINT),
+        types: {
+            ...KhalaTypes,
+            ...PhalaSDKTypes,
+        },
+    })
+    console.log(`Connected to node, create contract object`)
+    const executor = await createContract(nodeApi, PRUNTIME_ENDPOINT, contract, CONTRACT_ID)
+    console.log(`Contract objected created with contract ID: ${CONTRACT_ID}`)
+
+    const keyring = new Keyring({ type: 'sr25519' })
+    const alice = keyring.addFromUri('//Alice')
+    const certAlice = await PhalaSdk.signCertificate({
+        api: nodeApi,
+        pair: alice,
+    })
+
+    console.log(`Start query contract periodically...`)
+
+    // Trigger task search every 30 seconds
+    setInterval(async () => {
+        console.log(`🔍Trigger actived task search from ${SOURCE} for worker ${EXE_WORKER}`)
+        await executor.query.run(certAlice,
+            {},
+            {'Fetch': [SOURCE, EXE_WORKER]}
+        )
```

Comment:
Here are some real-world issues that I am aware of:
- Connection closed unexpectedly.
- Web socket connection hangs indefinitely (need to implement a timeout and query kill mechanism).
- Unexpected errors occur, and exceptions cause the process to crash.

A potentially improved version is available (tests are needed, and I have asked ChatGPT to handle that and not  yet verify the changes):

```javascript
const executeTask = async () => {
    try {
        console.log(`🐌Trigger task executing`);
        await Promise.race([
            executor.query.run(certAlice, {}, 'Execute'),
            new Promise((_, reject) => {
                setTimeout(() => {
                    reject(`Task execution timed out after ${timeout} milliseconds`);
                }, timeout);
            })
        ]);
        console.log(`Task executed successfully`);
    } catch (error) {
        console.error(`An error occurred during task execution: ${error}`);
    }
};

const startInterval = (intervalTime, timeout) => {
    setInterval(() => {
        executeTask(timeout);
    }, intervalTime);
};

startInterval(10000, 60000); // set the interval time to 10 seconds (10,000 milliseconds) and the task execution timeout to 1 minute (60,000 milliseconds)
```

Reply:
Thanks for sharing, consider the `shcudler.js` is only for demo purpose, I'm not going to handle them in here, probably fix it in future production shcduler